### ISSUE TITLE
CI: Simplify and improve performance for build

### ISSUE THREAD

**Issue by sehz** [infinyon/fluvio#1206]
This is tracking issue for improving build and release CI

- [x] https://github.com/infinyon/fluvio/issues/1205 (@nicholastmosher) 
- [x] https://github.com/infinyon/fluvio/issues/1207 @sehz @nicholastmosher 
- [x] Separate out Client builds/checks @nacardin 
- [x] Improve build time. @sehz 
- [x] Build Re-usable action for sccache. @sehz 
- [x] Use pre-build binaries for all test @sehz 
- [x] Move to Github runner for all test @sehz (deferred)
- [x] Move Upgrade Test to dev
- [x] Improve Upgrade Test Performance @tjtelan (deferred) 
- [x] Consolidate WASM build (deferred)
- [x] Create CLI build for Mac and cross-build tool @nacardin 
- [x] Move unit test out of Platform build and run in debug mode
- [x] Build binaries for Arm64, Arm32 (Raspberry Pi), @nacardin 
- [x] Ensure custom runner is using separate runner pool (deferred)
- [x] Run composite action on dev release @sehz (blocked by installer issue)
- [x] Fix Installer and Package issue @nicholastmosher

**Comment by sehz** [comment]
Before starting this work, here was results for CI run (debug)
https://github.com/infinyon/fluvio/runs/2952818569?check_suite_focus=true

For Ubuntu
```
Build and Test artifacts.  14m 32
Unit Tests.  6m27s

Run sccache --stop-server || true
13
Stopping sccache server...
14
Compile requests                    968
15
Compile requests executed           673
16
Cache hits                            0
17
Cache misses                        673
18
Cache misses (Rust)                 673
19
Cache timeouts                        0
20
Cache read errors                     0
21
Forced recaches                       0
22
Cache write errors                    0
23
Compilation failures                  0
24
Cache errors                          0
25
Non-cacheable compilations            0
26
Non-cacheable calls                 295
27
Non-compilation calls                 0
28
Unsupported compiler calls            0
29
Average cache write               0.000 s
30
Average cache read miss           1.181 s
31
Average cache read hit            0.000 s
32
Failed distributed compilations       0
33

34
Non-cacheable reasons:
35
crate-type                          155
36
incremental                         131
37
-                                     8
38
-E                                    1
39

40
Cache location                  Local disk: "/home/runner/.cache/sccache"
41
Cache size                          459 MiB
42
Max cache size                        2 GiB
```

```
Doc Tests.   12m26s
```

**Comment by sehz** [comment]
publishing CLI is broken.  It was broken far as: https://github.com/infinyon/fluvio/actions/runs/1012501059

**Comment by sehz** [comment]
move generating client issues to https://github.com/infinyon/fluvio/issues/1261

**Comment by sehz** [comment]
mark as done
